project/SparkBuild.scala

```diff
@@ -238,7 +238,16 @@ object SparkBuild extends PomBuild {
           "-Wconf:cat=other-match-analysis&site=org.apache.spark.sql.catalyst.catalog.SessionCatalog.lookupFunction.catalogFunction:wv",
           "-Wconf:cat=other-pure-statement&site=org.apache.spark.streaming.util.FileBasedWriteAheadLog.readAll.readFile:wv",
           "-Wconf:cat=other-pure-statement&site=org.apache.spark.scheduler.OutputCommitCoordinatorSuite.<local OutputCommitCoordinatorSuite>.futureAction:wv",
-          "-Wconf:cat=other-pure-statement&site=org.apache.spark.sql.streaming.sources.StreamingDataSourceV2Suite.testPositiveCase.\\$anonfun:wv"
+          "-Wconf:cat=other-pure-statement&site=org.apache.spark.sql.streaming.sources.StreamingDataSourceV2Suite.testPositiveCase.\\$anonfun:wv",
+          // SPARK-33775 Suppress compilation warnings that contain the following contents.
+          "-Wconf:msg=^(?=.*?method|value|type|object|trait|inheritance)(?=.*?deprecated)(?=.*?since 2.13).+$:s",
+          "-Wconf:msg=^(?=.*?Widening conversion from)(?=.*?is deprecated because it loses precision).+$:s",
+          "-Wconf:msg=Auto-application to \\`\\(\\)\\` is deprecated:s",
+          "-Wconf:msg=Unicode escapes in triple quoted strings are deprecated:s",
```

Comment:
Some of these we should maybe just fix, like how many unicode escape issues do we have?
We've already got rid of most of the symbol literal usages and view bounds.
The rest look OK, maybe better to suppress than fix. We have to remember to undo some of them eventually like paying attention to what's deprecated in 2.13.

Reply:
There are total 15 compilation warnings related to `Unicode escapes in triple quoted strings are deprecated` in core and sql module.

Address 8e03c13 remove this match, and SPARK-33801 use to track this issue

Reply:
There are only 3 compilation warnings related to `view bounds are deprecated` in `SequenceFileRDDFunctions` , Address d97e0a7 remove this match, and SPARK-33804 use to track this issue

Reply:
Address 3cd8174 remove msg match of `symbol literal is deprecated` and left TODO(SPARK-33805) to remind undo the corresponding deprecated usage suppression rule after fixed.